src/stretch_kinematics_plugin.cpp

```diff
@@ -0,0 +1,313 @@
+  {
+    RCLCPP_ERROR(LOGGER, "Seed state must have size %d instead of size %d\n", dimension_, ik_seed_state.size());
+    error_code.val = moveit_msgs::msg::MoveItErrorCodes::NO_IK_SOLUTION;
+    return false;
+  }
+
+  if (!consistency_limits.empty())
+  {
+    if (consistency_limits.size() != dimension_)
+    {
+      RCLCPP_ERROR(LOGGER,
+                   "Consistency limits must be empty or have size %d instead "
+                   "of size %d\n",
+                   dimension_, consistency_limits.size());
+      error_code.val = moveit_msgs::msg::MoveItErrorCodes::NO_IK_SOLUTION;
+      return false;
+    }
+  }
+
+  solution.resize(dimension_);
+
+  RCLCPP_DEBUG_STREAM(LOGGER, "searchPositionIK: Position request pose is "
+                                  << ik_pose.position.x << " " << ik_pose.position.y << " " << ik_pose.position.z << " "
+                                  << ik_pose.orientation.x << " " << ik_pose.orientation.y << " "
+                                  << ik_pose.orientation.z << " " << ik_pose.orientation.w);
+
+  // auto ik_seed_state_local = ik_seed_state;
+  std::vector<double> arm_jmg_solution(solution.begin(), std::next(solution.begin(), mobile_base_index_));
+  const auto arm_jmg_consistency_limits =
+      consistency_limits.empty() ?
+          std::vector<double>{} :
+          std::vector<double>(consistency_limits.cbegin(), std::next(consistency_limits.cbegin(), mobile_base_index_));
+
+  const std::vector<double> arm_jmg_ik_seed_state(ik_seed_state.cbegin(),
+                                                  std::next(ik_seed_state.cbegin(), mobile_base_index_));
+
+  const auto arm_ik_solver = arm_jmg_->getGroupKinematics().first.solver_instance_;
+  Eigen::Isometry3d eigen_ik_pose;
+  tf2::fromMsg(ik_pose, eigen_ik_pose);
+  solution = ik_seed_state;
+  // Notation:
+  // B_T_A: The homogenous transformation of A frame w.r.t. B frame
+  // arm_b: Base frame of the kinematics solver for the arm jmg
+  unsigned int attempt = 0;
+  do
+  {
+    ++attempt;
+    // First we solve for the arm only by fixing the mobile base
+    state_->setJointGroupPositions(mobile_base_jmg_, &solution[mobile_base_index_]);
+    state_->updateLinkTransforms();
+    // Convert the desired pose from the arm + mobile base solver base frame to the arm group base frame
+    const auto arm_b_T_ik_pose = tf2::toMsg(state_->getGlobalLinkTransform(arm_ik_solver->getBaseFrame()).inverse() *
+                                            state_->getGlobalLinkTransform(getBaseFrame()) * eigen_ik_pose);
+    const bool ik_valid = arm_ik_solver->searchPositionIK(arm_b_T_ik_pose, arm_jmg_ik_seed_state, timeout * 0.25,
+                                                          arm_jmg_consistency_limits, arm_jmg_solution,
+                                                          solution_callback, error_code, options);
+    if (ik_valid || options.return_approximate_solution)  // found acceptable solution
```

Comment:
even if options.return_approximate_solution is true, shouldn't it still update the ik solution for the base before returning?